test/e2e/persistent_volume/persistent_volume_test.go

```diff
@@ -0,0 +1,349 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package persistent_volume
+
+import (
+	"time"
+
+	"github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/onsi/ginkgo/v2"
+
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+)
+
+var kClient k8s.KubeCtl
+var restClient yunikorn.RClient
+var dev = "dev-" + common.RandSeq(5)
+
+var _ = ginkgo.BeforeSuite(func() {
+	// Initializing kubectl client
+	kClient = k8s.KubeCtl{}
+	Ω(kClient.SetClient()).To(gomega.BeNil())
+
+	// Initializing rest client
+	restClient = yunikorn.RClient{}
+	Ω(restClient).NotTo(gomega.BeNil())
+
+	yunikorn.EnsureYuniKornConfigsPresent()
+
+	// Create namespace
+	ginkgo.By("Create namespace " + dev)
+	ns, err := kClient.CreateNamespace(dev, nil)
+	Ω(err).NotTo(HaveOccurred())
+	Ω(ns.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	// Clean up
+	err := kClient.DeletePVCs(dev)
+	Ω(err).NotTo(HaveOccurred())
+
+	err = kClient.DeletePVs(dev)
+	Ω(err).NotTo(HaveOccurred())
+
+	ginkgo.By("Tearing down namespace: " + dev)
+	err = kClient.TearDownNamespace(dev)
+	Ω(err).NotTo(HaveOccurred())
```

Comment:
We should first try to delete all of them and only then check the errors. We can miss something if something fails.

Reply:
If there are PVs that exist before the test starts, deleting all of them may cause errors. I think it would be safer to only delete the PVs related to the test namespace.

Reply:
Ah sorry. What I meant is that it's a better approach to perform the delete operations and then check the results. If, for whatever reason, `kClient.DeletePVCs()` fails, the namespace won't be deleted.

Reply:
My idea:

```
err := kClient.DeletePVCs(dev)
err2 := kClient.DeletePVs(dev)
ginkgo.By("Tearing down namespace: " + dev)
err3 := kClient.TearDownNamespace(dev)

Ω(err).NotTo(HaveOccurred())
Ω(err2).NotTo(HaveOccurred())
Ω(err3).NotTo(HaveOccurred())
```